src/components/InfiniteScroll/InfiniteScroll.tsx

```diff
@@ -49,7 +49,7 @@ export const InfiniteScroll: React.FC<InfiniteScrollProps> = ({
 
     if (
       hasMore &&
-      wardRef.current!.getBoundingClientRect().top - window.innerHeight <
+      wardRef?.current!.getBoundingClientRect().top - window.innerHeight <
```

Comment:
every time I switch from home to another page it throw an error. I suspect it's because this function is dispatch one more time when the component is already unmounted